pkg/operator/hive/hive.go

```diff
@@ -88,9 +88,9 @@ func (r *ReconcileHiveConfig) deployHive(hLog log.FieldLogger, h resource.Helper
 		)
 	}
 
-	if dc := instance.Spec.DisabledControllers; len(dc) != 0 {
-		hiveContainer.Args = append(hiveContainer.Args, "--disabled-controllers", strings.Join(dc, ","))
-	}
+	// Always add clustersync to the list of disabled controllers since clustersync is running in a statefulset now.
+	disabledControllers := append(instance.Spec.DisabledControllers, "clustersync")
+	hiveContainer.Args = append(hiveContainer.Args, "--disabled-controllers", strings.Join(disabledControllers, ","))
```

Comment:
Same question here, feels like this could be hard coded in the yaml to avoid one avenue of confusion for people.

Reply:
I agree on the statefulset one, but on this one, I need to be able to add to the list of disabled controllers that's in hiveconfig.

If I add it to the yaml, then the hiveconfig disabled controllers will stomp on this and if clustersync isn't disabled in the hiveconfig disabled controllers list, then it would run.

I'm willing to change, it just seems wrong to do on this one.

Reply:
Ah that makes sense, sounds good to me.